src/engine/source/base/logging/logging.hpp

```diff
@@ -1,71 +1,103 @@
+                                                                       {"warning", spdlog::level::warn},
+                                                                       {"error", spdlog::level::err},
+                                                                       {"critical", spdlog::level::critical},
+                                                                       {"off", spdlog::level::off}};
+
 struct LoggingConfig
 {
-    const char* filePath = nullptr;
-    const char* header = nullptr;
-    LogLevel logLevel = LogLevel::Info;
-    int pollInterval = 5000 /*ns*/;
+    const char* filePath {DEFAULT_LOG_PATH};
+    // To know more about the format parameters, please see: https://github.com/gabime/spdlog/wiki/3.-Custom-formatting
+    const char* headerFormat {DEFAULT_LOG_HEADER};
+    const char* logLevel {DEFAULT_LOG_LEVEL};
+    const uint32_t flushInterval {DEFAULT_LOG_FLUSH_INTERVAL};     ///< Value in ms
+    const uint32_t dedicatedThreads {DEFAULT_LOG_THREADS};         ///< 0 means no dedicated threads,
+                                                                   ///< if one or more then logsQueueSize takes effect
+    const uint32_t logsQueueSize {DEFAULT_LOG_THREADS_QUEUE_SIZE}; ///< Logs queue size to be processed by the dedicated
+                                                                   ///< threads (has to be 1 or more)
 };
 
-static inline void loggingInit(LoggingConfig const& cfg)
+// TODO: This emulates a global variable to fasten the access to the "default" logger, it can be improved
+inline auto getDefaultLogger(void)
 {
-    if (cfg.filePath)
+    static auto defaultLogger = spdlog::get("default").get();
+    return defaultLogger;
+}
+
+// TODO: this is a simple, basic, implementation of how to configure the logger, this won't go to production
+static inline void loggingInit(LoggingConfig& cfg)
+{
+    const bool doTruncateFile {false};
+    try
     {
-        std::string path = cfg.filePath;
-        // TODO: check portability issues
-        if ("/dev/stdout" == path)
-        {
-            fmtlog::setLogFile(stdout, false);
-        }
-        else if ("/dev/stderr" == path)
+        if (0 < cfg.dedicatedThreads)
         {
-            fmtlog::setLogFile(stderr, false);
-        }
-        else
-        {
-            fmtlog::setLogFile(cfg.filePath, false);
+            // Here we set the amount of DEDICATED threads
+            spdlog::init_thread_pool(cfg.logsQueueSize, cfg.dedicatedThreads);
         }
+        spdlog::flush_every(std::chrono::milliseconds(cfg.flushInterval));
+
+        // Logger initialization ("default" is the logger name, it can be any custom name)
+        spdlog::basic_logger_mt("default", cfg.filePath, doTruncateFile);
     }
-    else
+    catch (const spdlog::spdlog_ex& ex)
     {
-        fmtlog::setLogFile(stderr, false);
+        std::cerr << "Log initialization failed: " << ex.what() << std::endl;
```

Comment:
We should raise an exception, or the one that is already thrown, and manage it on the main or module's run function.

Reply:
Ok, I'll take your suggestion